src/coord/src/coord.rs

```diff
@@ -5111,17 +5119,28 @@ pub mod fast_path_peek {
             // Endpoints for sending and receiving peek responses.
             let (rows_tx, rows_rx) = tokio::sync::mpsc::unbounded_channel();
 
+            // Generate unique UUID. Guaranteed to be unique to all pending peeks, there's a very
```

Comment:
The documentation doesn't go into a ton of detail about how unique each UUID is (https://crates.io/crates/uuid), they are 128 bits and it does say 
> The uniqueness property is not strictly guaranteed, however for all practical purposes, it can be assumed that an unintentional collision would be extremely unlikely.

So this comment and the `contains_key` check may be overkill.